### PR TITLE
 aws- add modify-subnet-atrribute action for aws subnet

### PR DESCRIPTION
**Pull request by shineyb**

### CODE REVIEW

File: c7n/resources/vpc.py [hunk 1]
```diff
@@ -2575,3 +2577,44 @@ def process(self, resources, event=None):
...
+                actions:
+                  - type: modify-subnet-attribute
+                    attributes:
+                      "MapPublicIpOnLaunch: false"
+                      "idle_timeout.timeout_seconds": 120
+    """
+    pdb.set_trace()
+    schema = type_schema(
+        "modify-subnet-attribute",
+        {
```
Comment (kapilt) [inline c7n/resources/vpc.py:2606]: afaicr atm, these need to be **{}

File: c7n/resources/vpc.py [hunk 2]
```diff
@@ -2575,3 +2577,44 @@ def process(self, resources, event=None):
...
+            'map_public_ip_on_launch': {'type': 'boolean'}
+        })
+
+    permissions = ("ec2:ModifySubentAttributes",)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('ec2')
+        breakpoint()
```
Comment (kapilt) [inline c7n/resources/vpc.py:2614]: stray re debug, again assuming the wip pr was for public comment re feedback.

File: c7n/resources/vpc.py [hunk 3]
```diff
@@ -2575,3 +2577,44 @@ def process(self, resources, event=None):
...
+        for sub in resources:
+            self.manager.retry(
+                client.modify_subnet_attribute,
+                SubId=sub.subnet_id, MapPublicIpOnLaunch={"Value": False}
```
Comment (kapilt) [inline c7n/resources/vpc.py:2618]: s/SubId/SubnetId

**Comment by kapilt** [comment]
thanks for the pr, made some comments before i realized it was still in a debug state.

File: c7n/resources/vpc.py [hunk 4]
```diff
@@ -2575,3 +2574,48 @@ def process(self, resources, event=None):
...
+        })
+
+    permissions = ("ec2:ModifySubentAttributes",)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('ec2')
+        map_public_ip_on_launch = self.data.get('map_public_ip_on_launch')
+        for r in resources:
+            client.modify_subnet_attribute(
+                SubnetId=r['SubnetId'], MapPublicIpOnLaunch={'Value': map_public_ip_on_launch})
```
Comment (PratMis) [inline c7n/resources/vpc.py:2620]: the api parameters should be pulled from the policy

Comment (shineyb) [reply c7n/resources/vpc.py:2620]: Yes I am getting it from the policy: 
map_public_ip_on_launch = self.data.get('map_public_ip_on_launch')

**Review by kapilt — APPROVED** [review]
i've gone ahead and refactored this, lgtm

**Pull request merged**